### ISSUE TITLE
No support for XML based drawables

### ISSUE THREAD

**Issue by alex2069** [bumptech/glide#350]
We have a placeholder shape drawable (or sometimes shown directly as `.load(R.drawable.abc)`) which fails to display. The onException listener is called, but e is null.

The logs contain only:

<pre>com.example.app D/skia﹕ --- SkImageDecoder::Factory returned null</pre>

Glide: 3.5.2
Tested on Genymotion v4.4.2

The drawable:

``` xml
<shape xmlns:android="http://schemas.android.com/apk/res/android"
       android:shape="rectangle">
    <solid android:color="#ffabcdef"/>
</shape>
```

**Comment by sjudd** [comment]
Not totally surprising, BitmapFactory will handle image based drawables, but not shapes or ripples etc. We may be able to fix this in 4.0, but probably not before.

**Comment by TWiStErRob** [comment]
Here's a list of currently known (and wanted non-bitmap drawables):
* [ ] Bitmap alias: #683
* [ ] NinePatch (`.9.png`): #786, #814
* [ ] Vector: #794
* [ ] Shapes: #350 (this issue)
* [ ] already in-memory Bitmap/Drawable: #122 (only loosely related, not duplicate)
